ArcMovies/ArcMovies/Views/MainPage.xaml

```diff
@@ -0,0 +1,103 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
+             xmlns:b="clr-namespace:Prism.Behaviors;assembly=Prism.Forms"
+             xmlns:c="clr-namespace:ArcMovies.Converters;assembly=ArcMovies"
+             xmlns:ffimageloading="clr-namespace:FFImageLoading.Forms;assembly=FFImageLoading.Forms"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             xmlns:prism="clr-namespace:Prism.Mvvm;assembly=Prism.Forms"
+             prism:ViewModelLocator.AutowireViewModel="True"
+             x:Class="ArcMovies.Views.MainPage"
+             Title="{Binding Title}">
+    <ContentPage.Resources>
+        <ResourceDictionary>
+            <c:ItemTappedEventArgsConverter x:Key="itemTappedEventArgsConverter" />
+            <c:ItemVisibilityEventArgsConverter x:Key="itemAppearingEventArgsConverter" />
+            <c:NegateBooleanConverter x:Key="inverter"/>
+        </ResourceDictionary>
+    </ContentPage.Resources>
+    <ContentPage.ToolbarItems>
+        <ToolbarItem             
+            AutomationId="Search"            
+            Text="Search"
+            Command="{Binding ShowSearchMoviesCommand}"/>
+    </ContentPage.ToolbarItems>
+    <ContentPage.Content>
+        <StackLayout>
+            <Frame IsVisible="{Binding IsConnected, Converter={StaticResource inverter}}">
+                <StackLayout>
+                    <Label
+                        Text="You are offline. Check your connection and try again."                        
+                        HorizontalTextAlignment="Center"
+                        VerticalTextAlignment="Center"/>
+                    <Button
+                        Text="Try again"                        
+                        Command="{Binding LoadUpcomingMoviesCommand}"
+                        HorizontalOptions="Center"
+                        VerticalOptions="Center"
+                        WidthRequest="100"
+                        BackgroundColor="{StaticResource Primary}"
+                        TextColor="White"/>
+                </StackLayout>
+            </Frame>
+            <ListView 
+                x:Name="ItemsListView"                
+                IsVisible="{Binding IsConnected}"
+                ItemsSource="{Binding Movies}"
+                HasUnevenRows="true"
+                SeparatorVisibility="None"
+                RefreshCommand="{Binding LoadUpcomingMoviesCommand}"
+                IsPullToRefreshEnabled="true"
+                IsRefreshing="{Binding IsBusy, Mode=OneWay}"
+                CachingStrategy="RecycleElement">
+                <ListView.ItemTemplate>
```

Comment:
It's nice to keep `DataTemplate` in a separate `xaml` file, it allows code reuse and it`s cleaner.